codegen/smithy-aws-swift-codegen/src/main/kotlin/software/amazon/smithy/aws/swift/codegen/PresignerGenerator.kt

```diff
@@ -0,0 +1,91 @@
+)
+
+class PresignerGenerator : SwiftIntegration {
+    override fun writeAdditionalFiles(ctx: CodegenContext, delegator: SwiftDelegator) {
+        val service = ctx.model.expectShape<ServiceShape>(ctx.settings.service)
+
+        if (!AWSSigningMiddleware.isSupportedAuthentication(ctx.model, service)) return
+        val presignOperations = service.allOperations
+            .map { ctx.model.expectShape<OperationShape>(it) }
+            .filter { operationShape -> operationShape.hasTrait(Presignable.ID) }
+            .map { operationShape ->
+                check(AWSSigningMiddleware.hasSigV4AuthScheme(ctx.model, service, operationShape)) { "Operation does not have valid auth trait" }
+                PresignableOperation(service.id.toString(), operationShape.id.toString())
+            }
+        presignOperations.forEach { presignableOperation ->
+            val op = ctx.model.expectShape<OperationShape>(presignableOperation.operationId)
+            val inputType = op.input.get().getName()
+            delegator.useFileWriter("${ctx.settings.moduleName}/models/$inputType+Presigner.swift") { writer ->
+                renderPresigner(writer, ctx, delegator, presignableOperation)
+            }
+        }
+    }
+
+    private fun renderPresigner(
+        writer: SwiftWriter,
+        ctx: CodegenContext,
+        delegator: SwiftDelegator,
+        presignableOperation: PresignableOperation
+    ) {
+        val serviceShape = ctx.model.expectShape<ServiceShape>(ctx.settings.service)
+        val protocolGenerator = ctx.protocolGenerator?.let { it } ?: run { return }
+        val protocolGeneratorContext = ctx.toProtocolGenerationContext(serviceShape, delegator)?.let { it } ?: run { return }
+
+        writer.addImport(AWSClientConfiguration)
+        writer.addImport(SdkHttpRequest)
+
+        val operationsIndex = OperationIndex.of(ctx.model)
+        val httpBindingResolver = protocolGenerator.getProtocolHttpBindingResolver(protocolGeneratorContext)
+
+        val op = ctx.model.expectShape<OperationShape>(presignableOperation.operationId)
+        val inputType = op.input.get().getName()
+        writer.openBlock("extension $inputType {", "}") {
+            writer.openBlock("public func presign(config: \$N) -> \$T {", "}", AWSClientConfiguration, SdkHttpRequest) {
+                writer.write("let serviceName = \"${ctx.settings.sdkId}\"")
+                writer.write("let input = self")
+                val operationStackName = "operation"
+                for (prop in protocolGenerator.httpProtocolCustomizable.getClientProperties()) {
+                    prop.addImportsAndDependencies(writer)
+                    prop.renderInstantiation(writer)
+                    prop.renderConfiguration(writer)
+                }
+
+                val generator = MiddlewareExecutionGenerator(protocolGeneratorContext, writer, httpBindingResolver, "content-type", protocolGenerator.httpProtocolCustomizable, operationStackName)
```

Comment:
Oops, this needs to be updated.

Reply:
cant you get default content type from the protocol itself?

Reply:
yep! sure can!